### PR TITLE
Add call dimension to metric for ShardSyncTask addSuccessAndLatency

### PR DESCRIPTION
**Pull request by Renjuju**
## Changes
* Add call dimension
* Dimension used to separate Time metric which is being double counted currently

### CODE REVIEW

File: amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/ShardSyncTask.java [hunk 1]
```diff
@@ -73,6 +75,11 @@ public TaskResult call() {
             log.error("Caught exception while sync'ing Kinesis shards and leases", e);
             exception = e;
         } finally {
+            MetricsUtil.addSuccessAndLatency(scope,
```
Comment (ashwing) [inline amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/ShardSyncTask.java]: what happens to the latency metric in metric decorator?

Comment (Renjuju) [reply amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/ShardSyncTask.java]: I didn't want to change the metric in the metric decorator as double counting counting time is due to the underlying Archiver metric factory. The metric emitted is currently ignored in the decorator

File: amazon-kinesis-client/src/main/java/software/amazon/kinesis/metrics/MetricsCollectingTaskDecorator.java [hunk 2]
```diff
@@ -52,6 +52,7 @@ public TaskResult call() {
         try {
             result = other.call();
         } finally {
+            MetricsUtil.addLatency(scope, other.getClass().getSimpleName(), startTimeMillis, MetricsLevel.SUMMARY);
```
Comment (Renjuju) [inline amazon-kinesis-client/src/main/java/software/amazon/kinesis/metrics/MetricsCollectingTaskDecorator.java:55]: method signature:

```java
    public static void addLatency(@NonNull final MetricsScope metricsScope, final String dimension,
                                  final long startTime, @NonNull final MetricsLevel metricsLevel)
```

Comment (ashwing) [reply amazon-kinesis-client/src/main/java/software/amazon/kinesis/metrics/MetricsCollectingTaskDecorator.java:55]: Can you post the new metrics emitted with this change and their value?

Comment (Renjuju) [reply amazon-kinesis-client/src/main/java/software/amazon/kinesis/metrics/MetricsCollectingTaskDecorator.java:55]: sure, will do

**Pull request closed without merging**